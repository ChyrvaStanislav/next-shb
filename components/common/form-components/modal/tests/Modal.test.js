import React, { useState } from 'react';
import { fireEvent, render } from '@testing-library/react';
import Modal from '../index';

const onClose = jest.fn();
const onDisableBody = jest.fn();
const onEnableBody = jest.fn();
const disableBody = () => {
  onDisableBody();
};
const enableBody = () => {
  onEnableBody();
};

jest.mock('../utils', () => ({
  disableBody, enableBody,
}));


const Component = (props) => {
  const [isOpen, setIsOpen] = useState(false);

  const handleClick = () => {
    setIsOpen(state => !state);
  };

  return (
    <>
      <button type="button" onClick={handleClick}>Click</button>
      <Modal open={isOpen} setOpen={setIsOpen} onClose={onClose} {...props}>
        <h1>Children</h1>
      </Modal>
    </>
  );
};

describe('Modal renders', () => {
  it('should Modal renders when closed', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});


describe('Modal functionality', () => {
  it('should Modal open', () => {
    const { getByText } = render(<Component />);
    const button = getByText('Click');
    fireEvent.click(button);

    expect(document.querySelector('.open')).toBeInTheDocument();
  });

  it('should Modal close', () => {
    const { getByText } = render(<Component />);
    const button = getByText('Click');
    fireEvent.click(button);
    fireEvent.click(button);

    expect(document.querySelector('.open')).not.toBeInTheDocument();
  });

  it('should onClose on section work', () => {
    const { getByText, getByTestId } = render(<Component />);
    const button = getByText('Click');
    const section = getByTestId('section');
    fireEvent.click(button);
    expect(document.querySelector('.open')).toBeInTheDocument();

    fireEvent.mouseDown(section);
    expect(document.querySelector('.open')).not.toBeInTheDocument();
    expect(onClose).toBeCalled();
  });
});


describe('Enable & Disable body', () => {
  it('should enableBody calls', () => {
    const { getByText } = render(<Component />);
    const button = getByText('Click');
    fireEvent.click(button);

    expect(onDisableBody).toBeCalled();
  });

  it('should disableBody calls', () => {
    const { getByText } = render(<Component />);
    const button = getByText('Click');
    fireEvent.click(button);
    fireEvent.click(button);

    expect(onEnableBody).toBeCalled();
  });
});
