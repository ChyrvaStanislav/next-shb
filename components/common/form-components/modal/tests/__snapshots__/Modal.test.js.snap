// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders should Modal renders when closed 1`] = `
<DocumentFragment>
  <button
    type="button"
  >
    Click
  </button>
  <section
    class="componentContainer close"
    data-testid="section"
  >
    <div
      class="slider"
    >
      <h1>
        Children
      </h1>
    </div>
  </section>
</DocumentFragment>
`;
